// Copyright (c) YugaByte, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software distributed under the License
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
// or implied.  See the License for the specific language governing permissions and limitations
// under the License.
//
syntax = "proto2";

package yb.docdb;

import "yb/common/common.proto";
import "yb/util/opid.proto";

option java_package = "org.yb.docdb";

message KeyValuePairPB {
  optional bytes key = 1;
  optional bytes value = 2;
  optional fixed64 external_hybrid_time = 3;
}

// A set of key/value pairs to be written into RocksDB.
message KeyValueWriteBatchPB {
  repeated KeyValuePairPB write_pairs = 1;
  optional TransactionMetadataPB transaction = 2;
  optional bool DEPRECATED_may_have_metadata = 3;
  // Used by serializable isolation transactions and row locking statements to store read intents.
  // In case of read-modify-write operation both read_pairs and write_pairs could present.
  repeated KeyValuePairPB read_pairs = 5;
  optional RowMarkType row_mark_type = 6;
}

message ConsensusFrontierPB {
  optional OpIdPB op_id = 1;
  optional fixed64 hybrid_time = 2;
  optional fixed64 history_cutoff = 3;
}
